mock.go

```diff
@@ -137,6 +138,19 @@ func (m *Mock) invoke(call *Call) {
 
 	invocation.Expectation = expected
 
+	if expected.stdin != nil {
+		// read all of stdin
+		buf, err := ioutil.ReadAll(call.Stdin)
+		if err != nil {
+			panic("error reading stdin")
+		}
+		// copy to Expectation
+		expected.readStdin = make([]byte, len(buf))
+		copy(expected.readStdin, buf)
```

Comment:
Copying into `expected.readStdin` instead of assigning so that we still own `buf`…